components/CheckListDetail.tsx

```diff
@@ -0,0 +1,48 @@
+import { HTMLAttributes, useEffect, useRef, useState } from "react";
+import Image from "next/image";
+import ic_checked from "@/assets/icons/checkbox_checked.svg";
+import ic_empty from "@/assets/icons//checkbox_empty.svg";
+import styles from "./CheckListDetail.module.css";
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
+  defaultIsChecked: boolean;
+  defaultValue: string;
+}
+
+export default function CheckListDetail({
+  className = "",
+  defaultIsChecked = false,
+  defaultValue = "",
+  ...props
+}: Props) {
+  const [value, setValue] = useState(defaultValue ?? "");
+  const [isChecked, setIsChecked] = useState(defaultIsChecked);
```

Comment:
💊 제안
defaultValue, defaultIsChecked와 같이 변수명을 다르게 지정하신 이유가 props와 내부 상태 변수명이 충돌하거나 혼동될 수 있어서라면, 아래와 같이 구조 분해 할당 시 별칭을 주는 방식으로 작성하셔도 됩니다!

```suggestion
interface Props extends HTMLAttributes<HTMLDivElement> {
  isChecked: boolean;
  value: string;
}

export default function CheckListDetail({
  className = "",
  isChecked:_isChecked= false,
  value:_value = "",
  ...props
}: Props) {
  const [value, setValue] = useState(_value ?? "");
  const [isChecked, setIsChecked] = useState(_isChecked);
```